pkg/client/patch.go

```diff
@@ -59,8 +63,29 @@ func ConstantPatch(patchType types.PatchType, data []byte) Patch {
 	return RawPatch(patchType, data)
 }
 
+type MergeFromWithOptimisticLock struct{}
+
+func (m MergeFromWithOptimisticLock) ApplyToMergeFrom(in *MergeFromOptions) {
+	in.OptimisticLock = true
+}
+
+// MergeFromOption is some configuration that modifies options for a merge-from patch data.
+type MergeFromOption interface {
+	// ApplyToMergeFrom applies this configuration to the given patch options.
+	ApplyToMergeFrom(*MergeFromOptions)
+}
```

Comment:
Could this just be a function type?

Reply:
This is following the same pattern you find in options.go

Reply:
Cool, I didn't see any other Options types here and wasn't familiar with that